src/app/static/src/app/store/modelOptions/mutations.ts

```diff
@@ -17,6 +17,11 @@ export const mutations: MutationTree<ModelOptionsState> & MetadataMutations = {
     },
 
     update(state: ModelOptionsState, payload: DynamicFormMeta) {
-        state.optionsFormMeta = payload;
+        const allControlSectionLabels = payload.controlSections.map(c => c.label);
```

Comment:
We don't want to fully overwrite the existing state with the api response, in case some values have been filled out already. So if a control section already exists we don't overwrite it, but simply add any missing control sections.